CMSgov/onemac
# OY2 22419: TE Status for CMS view

Story: https://qmacbis.atlassian.net/browse/OY2-22419
Endpoint: See github-actions bot comment

### Details
Temporary Extension requests are managed by a form submission in OneMAC.  In SEA Tool, however, they are noted as a flag on the package and are not packages themselves.

OneMAC originally treated Temporary Extension Requests fully as a component, and because they appear on the top-level of the Dashboard, they ARE components, but they are handled by CMS more like requests... similar to Withdrawal Requests.

CMS wishes to show a different package status on Temporary Extensions for the CMS users.  In looking ahead to see changes coming to the Withdrawal Requests (namely, they are turning them into a form submission), it seemed best to change the successful Temporary Extension Request status from "Submitted" to "TE Requested."  This allows us to separate the status displays between roles and describes better the interactions between State and CMS.

### Changes
- Added a new status "TE Requested"
- Updated the submitwaiverextension handling to assign the currentStatus of "TE Requested" to successful Temporary Extension submissions.
- updated getMyPackages to show the State user "Submitted" and the CMS user "-- --" when showing the "TE Requested" status on the Dashboard and details pages.
- created the migration script to convert old TE submissions to the new status
- updated the auditArray attribute with details of the migration

### Implementation Notes
- Status mappings are handled by the back end to prevent a savvy front-end user from seeing more than they should.

### Test Plan
1. Login as a State user
2. Navigate to Dashboard->Waivers tab
3. Verify that Temporary Extension packages are visible and show "Submitted" status.
4. Click into a Temporary Extension details page
5. Verify the status is "Submitted"
6. Return to dashboard and Submit a New Temporary Extension
7.  Verify the new Temporary Extension shows "Submitted" status
8. Logout
9. Login as CMS user
10. Navigate to Dashboard-> Waivers tab
11. Verify that Temporary Extension packages are visible and show "-- --" status.
12. Click into a Temporary Extension details page
13. Verify the status is "-- --"
